src/ai/backend/manager/services/resource_preset/types.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass, field
+from typing import Any, Optional, override
+
+from ai.backend.common.types import BinarySize, ResourceSlot
+from ai.backend.manager.types import Creator, PartialModifier, TriState
+
+
+@dataclass
+class ResourcePresetCreator(Creator):
+    name: str
+    resource_slots: ResourceSlot
+    shared_memory: Optional[str]
+    scaling_group_name: Optional[str]
+
+    def fields_to_store(self) -> dict[str, Any]:
+        return {
+            "name": self.name,
+            "resource_slots": self.resource_slots,
+            "shared_memory": BinarySize.from_str(self.shared_memory)
+            if self.shared_memory
+            else None,
+            "scaling_group_name": self.scaling_group_name,
+        }
+
+
+@dataclass
+class ResourcePresetModifier(PartialModifier):
+    resource_slots: TriState[ResourceSlot] = field(default_factory=TriState[ResourceSlot].nop)
+    name: TriState[str] = field(default_factory=TriState[str].nop)
```

Comment:
This should be `OptionalState`
```suggestion
    name: OptionalState[str] = field(default_factory=OptionalState[str].nop)
```